src/EventCell.js

```diff
@@ -145,14 +145,14 @@ class EventCell extends React.Component {
                   // 100ms is given to setTimeout so that it fires after a right-click event - AR Tue Oct 24 14:10:25 EDT 2017
                 }, 100);
               }}
-              onClick={e => onSelect(event, e)}
+              onClick={e => onSelect(event.data, e)}
               /*onDoubleClick={e => onDoubleClick(event, e)}*/
               onDoubleClick={this.handleEditing}
             >
-              <StyledEvent {...event.styles}>
+              <StyledEvent {...event.data.styles}>
                 <div className="rbc-event-content" title={title}>
                   {Event && !this.state.isEditingEventTitle ? (
-                    <Event event={event} title={title} />
+                    <Event event={event.data} title={title} />
```

Comment:
Did you change the Event type to have a data field?

Reply:
I needed to add metadata on the segment position but I didn't want to mutate the user event object, so I decided that event would be as follows:

event = {
  data: object, // user settable object, RBC should not mutate this in any way.
  ...metadata, // whatever metadata here, currently only position object, which holds segment position info
}

Reply:
I am thinking that any meta information should be placed in a meta tag.